### PR TITLE
Check if ImageRegistry installed

### PR DESCRIPTION
**Pull request by xiuwang**

### CODE REVIEW

**Comment by xiuwang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-basecap-none-f28

File: ci-operator/step-registry/baremetal/lab/agent/install/ephemeral/wait-for/baremetal-lab-agent-install-ephemeral-wait-for-commands.sh [hunk 1]
```diff
@@ -31,6 +31,17 @@ function get_ready_nodes_count() {
...
 
 function update_image_registry() {
+  # from OCP 4.14, the image-registry is optional, check if ImageRegistry capability is added
+  knownCaps=`oc get clusterversion version -o=jsonpath="{.status.capabilities.knownCapabilities}"`
+  if [[ ${knownCaps} =~ "ImageRegistry" ]]; then
+      echo "knownCapabilities contains ImageRegistry"
+      # check if ImageRegistry capability enabled
+      enabledCaps=`oc get clusterversion version -o=jsonpath="{.status.capabilities.enabledCapabilities}"`
+        if [[ ! ${enabledCaps} =~ "ImageRegistry" ]]; then
+            echo "ImageRegistry capability is not enabled, skip the following tests..."
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/install/ephemeral/wait-for/baremetal-lab-agent-install-ephemeral-wait-for-commands.sh:41]:
```suggestion
            echo "ImageRegistry capability is not enabled, skip the image registry configuration..."
```

**Review by aleskandro — COMMENTED** [review]
Hi @xiuwang, thanks for this PR. Just one note, that function is available in several files under the baremetal/ folder (they should all be like `*-install-*-commands.sh`. Can you change all in sync?

File: ci-operator/step-registry/baremetal/lab/agent/install/ephemeral/wait-for/baremetal-lab-agent-install-ephemeral-wait-for-commands.sh (hunk 1 above)
Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/install/ephemeral/wait-for/baremetal-lab-agent-install-ephemeral-wait-for-commands.sh:41]: this is not the step used by the IPI workflow

**Comment by xiuwang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-basecap-none-f28

**Comment by xiuwang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-basecap-none-f28

**Comment by xiuwang** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-basecap-none-f28

**Comment by aleskandro** [comment]
@xiuwang youd reach at the owners for aws and vsphere ir split into more PRs

**Comment by xiuwang** [comment]
@yunjiang29 @sgaoshang Could you help review the change for aws and vsphere platforms?
Thanks

**Comment by xiuwang** [comment]
@aleskandro Only updated BM related files now

**Comment by aleskandro** [comment]
Thanks @xiuwang!

**Pull request merged**